### ISSUE TITLE
LTS support: discrete Period grouping steps by time range mapping is not yet working, scrolling neither.

### ISSUE THREAD

**Issue by netweaver1970** [dbuezas/lovelace-plotly-graph-card#101]
Added as a new issue for tracking purposes and details updated where needed:

When specifying the discrete mapping steps in YAML, the period grouping is not shown any more.
In 'auto' mode, it shows up fine. It would be nice if it was there, to validate the correct behaviour. 
If you don't want to put it there by default (e.g. for space savings on the graph), please make it at least 'includable' with an option.
Or in upcoming beta releases included it, till all has been verified to be correct. Then it can be removed if you want.

So right now I can't really tell if the discrete grouping is indeed working. 
But already by looking at the graphs I'm expecting it doesn't ...

24hr time range, data point labels are per hour (should be 5minutes)
![image](https://user-images.githubusercontent.com/18028618/199233953-75f56554-37e0-4ab4-bd20-181b5bce167b.png)

1 month time range, data points labels are still per hour (should be day):
![image](https://user-images.githubusercontent.com/18028618/199234020-57056af5-c20d-46ca-90e5-4bdd3064b127.png)

1 year time frame, data points label per day but multiple data points shown for the SAME date, hence the zigzag. This still points to an hourly number of data points. So not a single mean period grouping by day has been used:
![image](https://user-images.githubusercontent.com/18028618/199234061-112f993c-936c-4abb-91a3-289df4bffdce.png)

I tried 1.6.0-beta and after reverting to 1.5.3, same behaviour.

Zooming in/out doesn't change any period value, too.

My YAML:

```
type: custom:plotly-graph
entities:
  - entity: sensor.th_livingroom_temperature
    statistic: mean
    legendgroup: 1
    name: Livingroom
    yaxes: y1
    line:
      color: red
      shape: spline
  - entity: sensor.th_kitchen_temperature
    statistic: mean
    legendgroup: 2
    name: Kitchen
    yaxes: y1
    line:
      color: orange
      shape: spline
  - entity: sensor.th_garage_temperature
    statistic: mean
    legendgroup: 3
    name: Garage
    yaxes: y1
    line:
      color: green
      shape: spline
  - entity: sensor.buienradar_temperature
    statistic: mean
    legendgroup: 4
    visible: legendonly
    name: Outside
    yaxes: y2
    line:
      color: gray
      shape: spline
  - entity: sensor.th_office2_temperature
    statistic: mean
    legendgroup: 5
    name: Office
    yaxes: y1
    line:
      dash: dot
      shape: spline
  - entity: sensor.th_guest_temperature
    legendgroup: 6
    name: Guest
    yaxes: y1
    line:
      dash: dot
      shape: spline
      color: red
  - entity: sensor.th_master_br_temperature
    statistic: mean
    legendgroup: 7
    name: Master
    yaxes: y1
    line:
      dash: dot
      shape: spline
      color: orange
  - entity: sensor.th_alastair_temperature
    statistic: mean
    legendgroup: 8
    name: Alastair
    yaxes: y1
    line:
      dash: dot
      shape: spline
  - entity: sensor.th_bathroom_temperature
    statistic: mean
    legendgroup: 9
    name: Bathroom
    yaxes: y1
    line:
      dash: dot
      shape: spline
hours_to_show: 24
no_default_layout: true
layout:
  dragmode: pan
  margin:
    t: 30
    l: 50
    r: 30
    b: 50
  showlegend: true
  height: 1028
  legend:
    x: 0
    'y': 1
    bgcolor: rgba(0,0,0,.4)
    orientation: h
    itemsizing: fixed
    font:
      size: 10
  grid:
    rows: 3
    columns: 1
    roworder: top to bottom
  yaxis:
    title:
      text: Temperature
  yaxis2:
    title:
      text: Humidity
  yaxis3:
    title:
      text: Abs. Humidity
  plot_bgcolor: black
  y1:
    gridcolor: whitesmoke
    zeroline: false
    fixedrange: true
  xaxis:
    rangeselector:
      'y': 1.02
      buttons:
        - count: 3
          step: day
        - count: 7
          step: day
        - count: 14
          step: day
        - count: 1
          step: month
        - count: 3
          step: month
        - count: 12
          step: month
        - count: 36
          step: month
defaults:
  entity:
    line: null
    period:
      24h: 5minute
      3d: 5minute
      7d: hour
      14d: hour
      1m: day
      3m: day
      12m: day
      36m: day
  width: 1
title: Climate
```

**Comment by dbuezas** [comment]
Uhm. I think I forgot to consider the defaults. Does it work if you put it in a specific entity?

**Comment by netweaver1970** [comment]
> Uhm. I think I forgot to consider the defaults. Does it work if you put it in a specific entity?

no. I tried it 1st on the entries itself. Then using the defaults. No go in both cases.
It seems to get somehow always the hour grouping period.

**Comment by netweaver1970** [comment]
> Regarding setting what `period: auto` means (like a default` i can add it too. Something like:
> 
> ```yaml
> custom_periods:
>   energy:
>     0: 5minutes #can be omitted
>     1w: day
>   temperature:
>       ...
> entities:
>   - entity: sensor.xxx
>     period: energy 
> ```

That would be even more flexible, that would mean a kind of 'mapping profiles', to avoid potential repeating manual mappings all the time. But maybe make the initial mvp work first, and only then get the different profiles in :)

**Comment by dbuezas** [comment]
Ok, I'll try to debug it tomorrow. It's odd, I thought i tested it well

**Comment by dbuezas** [comment]
It works fine. Did you test the branch I mentioned [here](https://github.com/dbuezas/lovelace-plotly-graph-card/issues/96#issuecomment-1298342964) or just the previously released version?

**Comment by dbuezas** [comment]
Anyway, the branch is released as v1.6.0 now

**Comment by netweaver1970** [comment]
Indeed, I was using the released version, not your separate branch. Mea culpa, didn't pay attention to the URL.

Now I'm on released version 1.6.1. And I see improvement but not there yet. Bottom 2 time ranges get correct period (5minutes), all others get period hours, while top 4 biggest time ranges should get days.

The selected period is now shown so easy to verify.

**Comment by netweaver1970** [comment]
I'm holding off testing and reporting bugs/strange behaviour till the auto range fixes from the offset branch are merged into the master branch and released. Otherwise, it's a moving target.
